src/pairs/GameState.py

```diff
@@ -22,36 +22,44 @@ def deepcopy(self):
     def next_draw_states(self):
         states = []
         for card_type in self.deck.curr_deck:
+            total_cards = sum(self.deck.curr_deck.values())
             if self.deck.curr_deck[card_type] > 0:
                 new_state = self.deepcopy()
                 new_state.deck.draw_specific_card(card_type)
                 new_state.players[0].add_card(card_type)
                 new_state.players.rotate(-1)
-                states.append(new_state)
+                states.append((new_state, self.deck.curr_deck[card_type] / total_cards))
         return states
 
     def next_fold_state(self):
         next_fold_state = self.deepcopy()
         minimum = min(
             min(card_type for card_type, freq in player.hand_state.items() if freq > 0)
             for player in next_fold_state.players)
+        if sum(minimum) == 0:
```

Comment:
You need to have this check before the minimum calculation to prevent the minimum of an empty sequence error. Alternatively, you can use a default for Python `min`:
https://docs.python.org/3/library/functions.html#min
```
minimum = min(
    min(card_type for card_type, freq in player.hand_state.items() if freq > 0, default = float('inf'))
    for player in next_fold_state.players, default=float('inf'))
if minimum == float('inf'):
    return None
```

Reply:
yup, done